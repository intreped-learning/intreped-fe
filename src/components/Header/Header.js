import React from 'react';
import './Header.scss';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

export const Header = () => {
  const dispatch = useDispatch();
  const { teacher } = useSelector(state => state)

  const handleLogOut = (e) => {
    e.preventDefault();
    dispatch({
      type: 'LOGOUT'
    })
  }
  return (
    <header>
      <Link className="logo" id="logo" to='/'>
        <h1>
          INTREP<span className="ed">ED</span>
        </h1>
      </Link>
      {!teacher.id &&
        <div>
          <button className="sign-in-btn" onClick={() => dispatch({ type: 'TOGGLE_MODAL' })}> Sign In </button>
          <button className="sign-up-btn"> Sign Up </button>
        </div>}
      {teacher.id &&
        <div className='signed-in-info'>
        <p className='teacher-name'>Welcome {teacher.name}!</p>
        <Link to='/dashboard' className='link-to-dashboard'>Dashboard</Link>
        <button className='sign-out-btn' onClick={handleLogOut}>Sign Out</button>
        </div>}
    </header>
  )
}
