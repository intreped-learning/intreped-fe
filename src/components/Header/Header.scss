header {
  background-color: #453B7F;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

h1 {
  font-family: Arial, Helvetica, sans-serif;
  color: #fff; 
  padding: 10px; 
  width: 15%;
  margin: 0;
  display: inline-block;
  justify-self: flex-start;
}

.ed {
  color: #8AE588;
}

#logo {
  text-decoration: none;
}

.sign-in-btn {
  color: #fff;
  background-color: transparent;
  border: none;
}

.sign-up-btn {
  background-color: #8AE588;
  border-radius: 30px;
  border: 1px solid transparent;
  margin: 15px;
  padding: 5px;
  height: 35px;
}