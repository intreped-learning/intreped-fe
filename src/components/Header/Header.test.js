import React from 'react';
import { Header } from './Header';
import { mount} from 'enzyme';

describe('Header', () => {
  it.skip('renders', () => {
    mount(<Header />);
  });
});