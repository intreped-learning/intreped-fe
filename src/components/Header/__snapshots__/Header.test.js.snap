// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header>
  <Link
    className="logo"
    id="logo"
    to="/"
  >
    <h1>
      INTREP
      <span
        className="ed"
      >
        ED
      </span>
    </h1>
  </Link>
  <div>
    <button
      className="sign-in-btn"
    >
       Sign In 
    </button>
    <button
      className="sign-up-btn"
    >
       Sign Up 
    </button>
  </div>
</header>
`;
